src/ucp/tag/tag_match.inl

```diff
@@ -212,29 +213,96 @@ ucp_tag_unexp_search(ucp_tag_match_t *tm, ucp_tag_t tag, uint64_t tag_mask,
     return NULL;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_request_recv_offload_data(ucp_request_t *req, const void *data,
+                              size_t length, size_t offset, unsigned recv_flags,
+                              int *last)
+{
+    ucs_status_t status = UCS_OK;
+    ucp_offload_last_ssend_hdr_t *s_hdr;
+
+    /* Should be used in multi-fragmented flow only */
+    ucs_assert(!(recv_flags & UCP_RECV_DESC_FLAG_EAGER_ONLY));
+
+    if (recv_flags & UCP_RECV_DESC_FLAG_EAGER_LAST) {
+        *last = 1;
```

Comment:
why need to return "last" to the caller, if it's defined solely by the input parameter recv_flags ?

Reply:
just to encapsulate offload handling in one func. Better move it out?

Reply:
yes